library/aes.c

```diff
@@ -1811,6 +1815,13 @@ int mbedtls_aes_self_test(int verbose)
                            (mode == MBEDTLS_AES_DECRYPT) ? "dec" : "enc");
         }
 
+#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH)
+        if (keybits > 128) {
+            mbedtls_printf("skipped\n");
+            continue;
+        }
+#endif /* MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH */
+
```

Comment:
Rather than add code to the self tests when `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` is specified, could we not set a different limit for the `for()`-loops?

```
#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH)
#define SELF_TEST_LOOP_LIMIT 2  /* AES-128 dec and enc only */
#else
#define SELF_TEST_LOOP_LIMIT 6  /* AES-128 dec & enc; AES-192 dec & enc; AES-256 dec & enc */
#endif
:
:
    for (i = 0; i < SELF_TEST_LOOP_LIMIT; i++) {
```

Reply:
make sense, will do it.

Reply:
> (it would also be great to remove the non-128-bit data for self-test, if not too difficult)

AFAIU, it's not difficult to remove the non-128-bit data. But we do need to add many
```
#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH)
#endif /* MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH */
```
pairs. Since, as for a single mode, plaintext and ciphertext are defined in a array for all `128-bit`, `192-bit` and `256-bit` cases.